cala-ledger-core-types/src/tx_template.rs

```diff
@@ -0,0 +1,75 @@
+use cala_cel_interpreter::{CelExpression, CelType, CelValue};
+use serde::{Deserialize, Serialize};
+
+use super::primitives::*;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TxTemplateValues {
+    pub id: TxTemplateId,
+    pub code: String,
+    pub description: Option<String>,
+    pub params: Option<Vec<ParamDefinition>>,
+    pub tx_input: TxInput,
+    pub entries: Vec<EntryInput>,
+    pub metadata: Option<serde_json::Value>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ParamDefinition {
+    pub name: String,
+    pub r#type: ParamDataType,
+    pub default: Option<CelExpression>,
+    pub description: Option<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EntryInput {
+    pub entry_type: CelExpression,
+    pub account_id: CelExpression,
+    pub layer: CelExpression,
+    pub direction: CelExpression,
+    pub units: CelExpression,
+    pub currency: CelExpression,
+    pub description: Option<CelExpression>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TxInput {
+    pub effective: String,
```

Comment:
Aren't these `CelExpression`s?

Reply:
Yeah you are right... will need to change this. thanks !

Reply:
Yeah you are right... will need to change this. thanks !